src/OpenCV/Features2d.hsc

```diff
@@ -285,6 +300,201 @@ orbDetectAndCompute orb img mbMask = unsafeWrapException $ do
+    fromPtr = objFromPtr SimpleBlobDetector $ \ptr ->
+                [CU.block| void {
+                  cv::Ptr<cv::SimpleBlobDetector> * simpleBlobDetector_ptr_ptr = $(Ptr_SimpleBlobDetector * ptr);
+                  simpleBlobDetector_ptr_ptr->release();
+                  delete simpleBlobDetector_ptr_ptr;
+                }|]
+
+data BlobFilterByArea
+     = BlobFilterByArea
+     { blob_minArea :: !Float
+     , blob_maxArea :: !Float
+     } deriving Eq
+
+data BlobFilterByCircularity
+     = BlobFilterByCircularity
+     { blob_minCircularity :: !Float
+     , blob_maxCircularity :: !Float
+     } deriving Eq
+
+data BlobFilterByColor
+     = BlobFilterByColor
+     { blob_blobColor :: !Word8
+     } deriving Eq
+
+data BlobFilterByConvexity
+     = BlobFilterByConvexity
+     { blob_minConvexity :: !Float
+     , blob_maxConvexity :: !Float
+     } deriving Eq
+
+data BlobFilterByInertia
+     = BlobFilterByInertia
+     { blob_minInertiaRatio :: !Float
+     , blob_maxInertiaRatio :: !Float
+     } deriving Eq
+
+data SimpleBlobDetectorParams
+   = SimpleBlobDetectorParams
+     { blob_minThreshold :: !Float
+     , blob_maxThreshold :: !Float
+     , blob_thresholdStep :: !Float
+     , blob_minRepeatability :: !Int32
+     , blob_minDistBetweenBlobs :: !Float
+     , blob_filterByArea :: !(Maybe BlobFilterByArea)
+       -- ^ Extracted blobs have an area between 'minArea' (inclusive) and
+       --   'maxArea' (exclusive).
+     , blob_filterByCircularity :: !(Maybe BlobFilterByCircularity)
+       -- ^ Extracted blobs have circularity
+       --   @(4 * pi * Area)/(perimeter * perimeter)@ between 'minCircularity'
+       --   (inclusive) and 'maxCircularity' (exclusive).
+     , blob_filterByColor :: !(Maybe BlobFilterByColor)
+       -- ^ This filter compares the intensity of a binary image at the center of
+       --   a blob to 'blobColor'. If they differ, the blob is filtered out. Use
+       --   @blobColor = 0@ to extract dark blobs and @blobColor = 255@ to extract
+       --   light blobs.
+     , blob_filterByConvexity :: !(Maybe BlobFilterByConvexity)
+       -- ^ Extracted blobs have convexity (area / area of blob convex hull) between
+       --   'minConvexity' (inclusive) and 'maxConvexity' (exclusive).
+     , blob_filterByInertia :: !(Maybe BlobFilterByInertia)
+       -- ^ Extracted blobs have this ratio between 'minInertiaRatio' (inclusive)
+       --   and 'maxInertiaRatio' (exclusive).
+     }
+
+defaultSimpleBlobDetectorParams :: SimpleBlobDetectorParams
+defaultSimpleBlobDetectorParams =
+    SimpleBlobDetectorParams
+    { blob_minThreshold = 50
+    , blob_maxThreshold = 220
+    , blob_thresholdStep = 10
+    , blob_minRepeatability = 2
+    , blob_minDistBetweenBlobs = 10
+    , blob_filterByArea = Just (BlobFilterByArea 25 5000)
+    , blob_filterByCircularity = Nothing
+    , blob_filterByColor = Just (BlobFilterByColor 0)
+    , blob_filterByConvexity = Just (BlobFilterByConvexity 0.95 (1/0))
```

Comment:
Perhaps bind `1/0` to some variable called `infinity :: Float`. I thought this was an error until I typed it into GHCI ;-) Should be top level definition local to this module.